### PR TITLE
Major changes and updates to facemapSVD pipeline

### PR DESCRIPTION
**Pull request by ttngu207**
- Rename SVD component-related table attributes
- Update ingestion to handle FullSVD analysis
- Remove redundant helper functions

### CODE REVIEW

File: element_facemap/facial_behavior_estimation.py [hunk 1]
```diff
@@ -301,33 +300,37 @@ def make(self, key):
...
+            def run_facemap_process():
+                facemap_run(
+                    filenames=video_files,
+                    savepath=output_dir.as_posix(),
+                    **params,
+                )
 
-        _, creation_time = get_loader_result(key, FacemapTask)
-        key = {**key, "processing_time": creation_time}
+            run_facemap_process()
```
Comment (MilagrosMarin) [inline element_facemap/facial_behavior_estimation.py]: I would suggest to make the function private and rename it to `_run_facemap_process()`.

File: setup.py [hunk 2]
```diff
@@ -28,9 +28,10 @@
     install_requires=[
         "datajoint>=0.13.0",
         "ipykernel>=6.0.1",
+        "scikit-learn==1.4.*",
         "opencv-python",
         "element-interface @ git+https://github.com/datajoint/element-interface.git",
-        "facemap @ git+https://github.com/kushalbakshi/facemap.git",
+        "facemap[gui] @ git+https://github.com/datajoint/facemap.git",
```
Comment (MilagrosMarin) [inline setup.py]: Following other elements, such as [`element-deeplabcut` setup](https://github.com/datajoint/element-deeplabcut/blob/main/setup.py), I would move the installation of the GUI to `extras_require` and make it optional.

**Review by MilagrosMarin — CHANGES_REQUESTED** [review]

File: element_facemap/facial_behavior_estimation.py (hunk 1 above)
Comment (MilagrosMarin) [reply element_facemap/facial_behavior_estimation.py]: An example [here](https://github.com/datajoint/element-array-ephys/blob/b510cb8aa5499d3f66adae1aa9a168c25303ac19/element_array_ephys/spike_sorting/si_spike_sorting.py#L208-L224), `_run_sorter()` in `element-array-ephys`.

**Review by kushalbakshi — COMMENTED** [review]
Looks good to me. If you could update the `run_facemap_process` function based on Milagros' comment above, it will ensure consistency in how we deploy the `memoized_result` functionality across Elements. Otherwise, this is ready to merge.

**Comment by ttngu207** [comment]
> Looks good to me. If you could update the `run_facemap_process` function based on Milagros' comment above, it will ensure consistency in how we deploy the `memoized_result` functionality across Elements. Otherwise, this is ready to merge.

@MilagrosMarin @kushalbakshi new commit added to address your comment.
There's one inconsistency in this soft convention we have here, in `element-deeplabcut`, we name the function `do_analyze_videos` (not `_do_analyze_videos`)
I'm fine either way, this is a minor point, it doesn't affect any functionality - it's good to stay consistent

**Comment by kushalbakshi** [comment]
> > Looks good to me. If you could update the `run_facemap_process` function based on Milagros' comment above, it will ensure consistency in how we deploy the `memoized_result` functionality across Elements. Otherwise, this is ready to merge.
> 
> @MilagrosMarin @kushalbakshi new commit added to address your comment. There's one inconsistency in this soft convention we have here, in `element-deeplabcut`, we name the function `do_analyze_videos` (not `_do_analyze_videos`) I'm fine either way, this is a minor point, it doesn't affect any functionality - it's good to stay consistent

Yes, it's probably a good idea to update that one too. As a general rule, any function within the make should probably follow the private function convention just to discourage its use outside the `make`

**Review by kushalbakshi — APPROVED** [review]

**Review by MilagrosMarin — APPROVED** [review]

**Pull request merged**